ui/packages/catalog-ui-search/src/main/webapp/react-component/select-coordinates/select-coordinates.tsx

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import styled from 'styled-components'
+import { hot } from 'react-hot-loader'
+import Dropdown from '../presentation/dropdown'
+import MenuAction from '../menu-action'
+import NavigationBehavior from '../navigation-behavior'
+import { ContextType } from '../presentation/dropdown'
+
+const Clipboard = require('clipboard')
+const announcement = require('component/announcement')
+
+type Props = {
```

Comment:
This component was pretty much copied from the `CopyCoordinates` component, so there's a lot of duplicate code. The only differences are the additional `getDistanceText()`, `coordHandler()`, and `clearHandler()` functions and the changed menu action child components. I'm not sure if there's a way to extend an existing React component, but if that's possible it would allow for a lot of code reuse.